### ISSUE TITLE
Extend `yarn infra x` to help recovery from lock-state

### ISSUE THREAD

**Issue by chrismilleruk** [goldstack/goldstack#101]
After terminating a `terraform` process which had hung, tf-state was left in a locked state.

This exists to ensure no more than one concurrent terraform user can be operating on the environment at a time.

[force-unlock](https://www.terraform.io/cli/commands/force-unlock) can be used to remedy this. It needs to be run twice, once to retrieve the lock-id and once again to unlock it. 

This is not easily done via `yarn infra ..` currently.

In addition, or alternatively, a generic pass-through command might be useful (e.g. `yarn infra terraform --version`)

**Comment by mxro** [comment]
That's a great idea! I think especially the pass through would be great to fulfil a number of diffrent use cases.

**Comment by mxro** [comment]
Implemented a first version of this: See the [draft of documentation for the new command](https://github.com/goldstack/goldstack/pull/117/files#diff-4f50633ec2b2da00d9407963a8a1deaa19c472008c1d56cf3b73b4a97c77072fR53)

Unfortunately Terraform is a bit unforgiving about provided command line arguments that are not used. Thus I had to add a few arguments controling which arguments from the package context are passed down to the Terraform command run (such as the variables). Also needed to enquire about the workspace that the command needs to apply to, thus it ended up being `yarn infra terraform [deployment] [...options] command`.

![image](https://user-images.githubusercontent.com/1448524/155868812-0381a660-196e-4e9e-94c7-16310761aa26.png)

**Comment by chrismilleruk** [comment]
I pushed an update to this branch which intends to solve https://github.com/goldstack/goldstack/security/code-scanning/39

It's better from a security POV but may be less forgiving. lmk your thoughts.

**Comment by mxro** [comment]
Love it! I think using `execFileSync` is a great idea.

I think generally for now these commands should definitely not be run on a server based on user input. But only on developer machines and in build pipelines, since no sanitation etc. has been performed yet!

Will merge in and release this in a new version. Also raised issue #123

**Comment by mxro** [comment]
Looks like the new command breaks some of the end-to-end tests:

https://github.com/goldstack/goldstack/runs/5413411460?check_suite_focus=true#step:6:4036

```
Command failed: docker [
  'run',
  '--rm',
  '-v',
  '/tmp/set/project-nextjsbootstrap1/packages/app-nextjs-bootstrap-1/infra/aws:/app',
  '-e',
  'AWS_ACCESS_KEY_ID=***',
  '-e',
  'AWS_SECRET_ACCESS_KEY=***',
  '-e',
  'AWS_SESSION_TOKEN=',
  '-e',
  'AWS_DEFAULT_REGION=us-west-2',
  '-w',
  '/app',
  'hashicorp/terraform:1.1.3',
  'workspace',
  'select',
  'prod'
```

Looks like it did not pick up the Terraform variables correctly. I may not get to it today but will definitely have a look in the weekend!

**Comment by mxro** [comment]
Released with [v0.3.18](https://github.com/goldstack/goldstack/releases/tag/v0.3.18) and ready for verification. As per PR linked above, I reverted changes related to `execSafe` for now.

**Comment by chrismilleruk** [comment]
Thanks @mxro this is working well. Confirmed with latest.
